src/main/java/org/prebid/server/bidder/huaweiads/HuaweiUtilRequest.java

```diff
@@ -0,0 +1,474 @@
+                int w = image.getW() != null ? image.getW() : 0;
+                int wMin = image.getWmin() != null ? image.getWmin() : 0;
+                int hMin = image.getHmin() != null ? image.getHmin() : 0;
+
+                if (h != 0 && w != 0) {
+                    width = w;
+                } else if (wMin != 0 && hMin != 0) {
+                    width = wMin;
+                }
+            }
+        }
+        return width;
+    }
+
+    private static Integer resolveNativeHeight(List<Asset> assets) {
+        if (assets == null) {
+            return null;
+        }
+        Integer height = null;
+        for (Asset asset : assets) {
+            ImageObject image = asset.getImg();
+            if (image != null) {
+                int h = image.getH() != null ? image.getH() : 0;
+                int w = image.getW() != null ? image.getW() : 0;
+                int wMin = image.getWmin() != null ? image.getWmin() : 0;
+                int hMin = image.getHmin() != null ? image.getHmin() : 0;
+
+                if (h != 0 && w != 0) {
+                    height = h;
+                } else if (wMin != 0 && hMin != 0) {
+                    height = hMin;
+                }
+            }
+        }
+        return height;
+    }
+
+    private static int convertAdtypeString2Integer(String adType) {
+        switch (adType) {
+            case "banner":
+                return BidTypes.BANNER_CODE.getValue();
+            case "native":
+                return BidTypes.NATIVE_CODE.getValue();
+            case "rewarded":
+                return BidTypes.REWARDED_CODE.getValue();
+            case "splash":
+                return BidTypes.SPLASH_CODE.getValue();
+            case "interstitial":
+                return BidTypes.INTERSTITIAL_CODE.getValue();
+            case "roll":
+                return BidTypes.ROLL_CODE.getValue();
+            default:
+                return BidTypes.BANNER_CODE.getValue();
+        }
+    }
+
+    private static Integer resolveTotalDuration(Imp imp, String lowerAdType) {
+        if (lowerAdType.equals("roll")) {
+            final Video video = imp.getVideo();
+            if (video != null && video.getMaxduration() != null && video.getMaxduration() >= 0) {
+                return video.getMaxduration();
+            } else {
+                throw new PreBidException("resolveTotalDuration: Video maxDuration is empty when adtype is roll");
+            }
+        }
+        return null;
+    }
+
+    private static Integer resolveBannerWidth(Banner banner) {
+        if (banner.getW() != null && banner.getH() != null) {
+            return banner.getW();
+        }
+        return 0;
+    }
+
+    private static Integer resolveBannerHeight(Banner banner) {
+        if (banner.getW() != null && banner.getH() != null) {
+            return banner.getH();
+        }
+        return null;
+    }
+
+    private static List<HuaweiFormat> resolveFormat(Banner banner) {
```

Comment:
it is better to return empty list instead of null, so you can just return `CollectionUtils.emptyIfNull(banner.getFormat()).stream()` without condition, and then work safely with this collection on top level